editor.planx.uk/src/@planx/components/FindProperty/model.ts

```diff
@@ -1,35 +1,62 @@
+import type { SchemaOf } from "yup";
+import { object, string } from "yup";
+
 import { MoreInformation, parseMoreInformation } from "../shared";
 
 export interface FindProperty extends MoreInformation {
   title: string;
   description: string;
+  allowNewAddresses?: boolean;
+  newAddressTitle?: string;
+  newAddressDescription?: string;
 }
 
 export const parseFindProperty = (
   data: Record<string, any> | undefined
 ): FindProperty => ({
   title: data?.title || "",
   description: data?.description || "",
+  allowNewAddresses: data?.allowNewAddresses || false,
+  newAddressTitle: data?.newAddressTitle || "",
+  newAddressDescription: data?.newAddressDescription || "",
   ...parseMoreInformation(data),
 });
 
-export interface SiteAddress {
-  uprn: string;
-  blpu_code: string;
+// Minimum-required address details if proposing a new non-UPRN address
+//   these fields also satisfy component dependencies like DrawBoundary & PlanningConstraints
+export interface MinimumSiteAddress {
   latitude: number;
   longitude: number;
-  organisation: string | null;
-  sao: string | null;
-  pao: string;
-  street: string;
-  town: string;
-  postcode: string;
   x: number;
   y: number;
-  planx_description: string;
-  planx_value: string;
-  single_line_address: string;
   title: string;
+  source: "os" | "proposed";
+}
+
+// Full SiteAddress reflects selecting a known address from the OS Places API "LPI" datasource
+export interface SiteAddress extends MinimumSiteAddress {
+  uprn?: string;
+  blpu_code?: string;
+  organisation?: string | null;
+  sao?: string | null;
+  pao?: string;
+  street?: string;
+  town?: string;
+  postcode?: string;
+  single_line_address?: string;
+  planx_description?: string; // joined via table blpu_codes
+  planx_value?: string; // joined via table blpu_codes
 }
 
+export type NewAddressInputs = {
```

Comment:
nit: `UserDefinedAddress`? "New" could mean a few things to me.

Not a strongly held opinion!

Reply:
very fair point - updating to `ProposedAddress` as that best aligns to the new `_address.source: "os" | "proposed"` reference field I've also introduced. does that seem more clear?

Reply:
Perfect 👍